### PR TITLE
ANC Logic - diagnosis HIV + Malaria infrastructure

### PR DESCRIPTION
**Pull request by anvmn**
#136

### CODE REVIEW

**Comment by anvmn** [comment]
@mariano-dagostino please review

**Comment by anvmn** [comment]
Peek 2022-02-27 23-03
Peek 2022-02-28 20-36

File: client/src/elm/Backend/Update.elm [hunk 1]
```diff
@@ -3919,11 +3924,23 @@ generatePrenatalAssessmentMsgs currentDate language isChw activePage updateAsses
...
+
+                    _ =
+                        Debug.log "diagnosesAfter" diagnosesAfter
+
+                    _ =
+                        Debug.log "addedDiagnoses" addedDiagnoses
+
+                    _ =
+                        Debug.log "initialEncounterMsgs" initialEncounterMsgs
+
```
Comment (mariano-dagostino) [inline client/src/elm/Backend/Update.elm]: @anvmn Probably can be deleted, right?

File: client/src/elm/Backend/Update.elm [hunk 2]
```diff
@@ -3935,76 +3952,103 @@ generatePrenatalAssessmentMsgs currentDate language isChw activePage updateAsses
...
-                                        |> String.join ", "
+                                else
+                                    case addedDiagnoses of
+                                        [] ->
+                                            []
 
-                                message =
-                                    if isChw then
-                                        translate language Translate.DangerSignsLabelForChw ++ ": " ++ signs
+                                        [ DiagnosisPrescribeMebendezole ] ->
```
Comment (mariano-dagostino) [inline client/src/elm/Backend/Update.elm:3965]: @anvmn This seems an special case between `[]` and `[ first::rest ]` do we have any relevant comment to add here?

File: client/src/elm/Pages/PrenatalActivity/Update.elm [hunk 3]
```diff
@@ -2706,6 +2738,21 @@ update currentDate id db msg model =
...
+                    Debug.log "medication" medication
+
+                _ =
+                    Debug.log "reason" reason
+
+                _ =
+                    Debug.log "updatedValue" updatedValue
+
+                _ =
+                    Debug.log "updatedNonAdministrationSigns" updatedNonAdministrationSigns
```
Comment (mariano-dagostino) [inline client/src/elm/Pages/PrenatalActivity/Update.elm]: @anvmn Old debug code?

**Review by mariano-dagostino — CHANGES_REQUESTED** [review]
@anvmn Finished my review,

File: client/src/elm/Backend/Update.elm (hunk 1 above)
Comment (anvmn) [reply client/src/elm/Backend/Update.elm]: @mariano-dagostino well, as this is ongoing development, and more and more diagnoses are added, these logs are useful untill feature is completed.
However, to keep the code 'formal', I'll delete them.

File: client/src/elm/Backend/Update.elm (hunk 2 above)
Comment (anvmn) [reply client/src/elm/Backend/Update.elm:3965]: @mariano-dagostino Right. Added comments.

**Comment by anvmn** [comment]
@mariano-dagostino All comments addressed.

**Review by mariano-dagostino — APPROVED** [review]

**Pull request merged**